torch/csrc/jit/interpreter.cpp

```diff
@@ -216,46 +489,65 @@ Operation getOperation(jit::Node *node) {
     }
   IR_ELSEIF(FusionGroup)
     auto fusion_fn = sharedFusionCompiler().getOrCompile(value);
-    return [fusion_fn](const list_of_retainable & inputs, list_of_retainable & outputs) {
+    auto num_inputs = value->inputs().size();
+    return [fusion_fn, num_inputs](Stack & stack) {
       autograd::profiler::RecordFunction record("FusionGroup");
-      tensor_list tinputs = unsafeToTensorListShare(inputs);
-      tensor_list toutputs;
-      fusion_fn->launch(tinputs, toutputs);
-      moveToListOfRetainables(std::move(toutputs), outputs);
+      Stack toutputs;
+      // TODO: have fusion_fn work off of a stack as well
+      fusion_fn->launch(last(stack, num_inputs), toutputs);
+      drop(stack, num_inputs);
+      stack.insert(stack.end(), toutputs.begin(), toutputs.end());
+      return 0;
     };
   IR_ELSEIF(Constant)
     auto t = value->t(kvalue);
-    return [t](const list_of_retainable & inputs, list_of_retainable & outputs) {
-      outputs.push_back(toRetainableShare(t));
+    return [t](Stack & stack) {
+      stack.push_back(t);
```

Comment:
I had to make this
```
      stack.push_back(torch::autograd::make_variable(t, true));
```

to get this to work when passing in Variables from python

Reply:
Looks like that broke other stuff. @zdevito let's discuss this when you get a chance